### PR TITLE
View application page tests

### PR DESCRIPTION
**Pull request by 1JW1**
##What

Added tests to the view applicant page

## Why

To test that the page behaves as expected and increase code coverage for the application

### CODE REVIEW

File: components/admin/personal-details.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export default function PersonalDetails({
             <td className="govuk-table__cell">
               <ul className="lbh-list lbh-list--compressed">
                 <li>
-                  <strong>
+                  <strong data-testid="applicant-name">
```
Comment (sam-drumm) [inline components/admin/personal-details.tsx:37]: for consistency this should be prefixed with test

File: components/admin/personal-details.tsx [hunk 2]
```diff
@@ -55,7 +55,7 @@ export default function PersonalDetails({
                     <Link
                       href={`mailto:${applicant.contactInformation.emailAddress}`}
                     >
-                      <a className="lbh-link">
+                      <a className="lbh-link" data-testid={'applicant-email'}>
```
Comment (sam-drumm) [inline components/admin/personal-details.tsx:58]: for consistency, this should be prefixed with test

File: components/summary-list.tsx [hunk 3]
```diff
@@ -56,7 +56,11 @@ interface SummaryListValueProps {
 export function SummaryListValue({
   children,
 }: SummaryListValueProps): JSX.Element {
-  return <dd className="govuk-summary-list__value">{children}</dd>;
+  return (
+    <dd className="govuk-summary-list__value" data-testid="activity-history">
```
Comment (sam-drumm) [inline components/summary-list.tsx:60]: test prefix

File: cypress/e2e/pages/viewApplication.cy.ts [hunk 4]
```diff
@@ -69,4 +69,75 @@ describe('View a resident application', () => {
...
+  it('allows readonly users to see note section', () => {
+    const application = generateApplication(applicationId, personId);
+    //ensure application requires assessment
+    application.status = ApplicationStatus.SUBMITTED;
+
+    cy.loginAsUser('readOnly');
+    cy.mockActivityHistoryApiEmptyResponse(applicationId, true);
+    cy.mockHousingRegisterApiGetApplications(applicationId, application, true);
+    ViewApplicationPage.visit(applicationId);
+    ViewApplicationPage.getNotesAndHistorySection()
```
Comment (sam-drumm) [inline cypress/e2e/pages/viewApplication.cy.ts:97]: Just thinking - as you're not writing in here anymore maybe enough to get it and make sure it exists?

File: cypress/e2e/pages/viewApplication.cy.ts [hunk 5]
```diff
@@ -69,4 +69,75 @@ describe('View a resident application', () => {
...
+      .click()
+      .get('[data-testid="add-text-input"]')
```
Comment (sam-drumm) [inline cypress/e2e/pages/viewApplication.cy.ts:99]: if you do need the text input id selector it should probably be in the page model. I'm pretty sure if you look in the development branch you'll see a components page model that will point to this element.

File: cypress/e2e/pages/viewApplication.cy.ts [hunk 6]
```diff
@@ -69,4 +69,75 @@ describe('View a resident application', () => {
...
+
+    cy.loginAsUser('readOnly');
+    cy.mockActivityHistoryApiEmptyResponse(applicationId, true);
+    cy.mockHousingRegisterApiGetApplications(applicationId, application, true);
+
+    ViewApplicationPage.visit(applicationId);
+    ViewApplicationPage.getNotesAndHistorySection()
+      .click()
+      .get('[data-testid="activity-history"]')
+      .should('exist');
```
Comment (sam-drumm) [inline cypress/e2e/pages/viewApplication.cy.ts:116]: I guess same as above for any of these - no need to click or get if you're just checking if it exits.

File: cypress/support/commands.ts [hunk 7]
```diff
@@ -73,7 +73,28 @@ Cypress.Commands.add(
...
+              _activityType: 'biddingNumberChangedByUser',
+              'assessment.biddingNumber': '1000001',
+            },
+            authorDetails: {
+              fullName: 'John Jones',
+              email: 'test@hackney.gov.uk',
+            },
+            sourceDomain: 'HousingRegister',
+          },
+        ],
```
Comment (sam-drumm) [inline cypress/support/commands.ts:97]: I think you'll need to adjust this, as it's currently used across the tests with empty response.

So you could just adjust it so you can pass a result to it, or otherwise it will be an empty object.

**Review by sam-drumm — COMMENTED** [review]
A few comments

File: components/admin/personal-details.tsx (hunk 1 above)
Comment (sam-drumm) [reply components/admin/personal-details.tsx:37]: ☝️

File: cypress/e2e/pages/viewApplication.cy.ts [hunk 8]
```diff
@@ -69,4 +89,73 @@ describe('View a resident application', () => {
...
+    const application = generateApplication(applicationId, personId);
+    //ensure application requires assessment
+    application.status = ApplicationStatus.SUBMITTED;
+
+    cy.loginAsUser('readOnly');
+    cy.mockActivityHistoryApiEmptyResponse(applicationId, true);
+    cy.mockHousingRegisterApiGetApplications(applicationId, application, true);
+
+    ViewApplicationPage.visit(applicationId);
+    cy.get('[data-testid="applicant-name"]').should(
```
Comment (sam-drumm) [inline cypress/e2e/pages/viewApplication.cy.ts:151]: ☝️😀

**Review by sam-drumm — APPROVED** [review]
